lib/PAUSE/package.pm

```diff
@@ -603,143 +489,131 @@ has the same version number and the distro has a more recent modification time.}
-Please report the case to the PAUSE admins at modules\@perl.org.},
-              );
-          $ok = 0;
-      }
-  }
+  my ($pkg_recs) = $dbh->selectall_arrayref(
+      qq{
+          SELECT package, version, dist
+          FROM packages
+          WHERE lc_package = ?
+      },
+      { Slice => {} },
+      lc $package,
+  );
 
-  return unless $self->_version_ok($pp, $package, $dist);
+  if (@$pkg_recs > 1) {
+      $Logger->log([
+          "conflicting records exist in packages table, won't index: %s",
+          [ @$pkg_recs ],
+      ]);
 
+      $ctx->abort_indexing_package($self, PKGERROR('db_conflict'));
+  }
 
-  if ($ok) {
-      my $query = qq{
-        UPDATE  packages
-        SET     package = ?, lc_package = ?, version = ?, dist = ?, file = ?,
-                filemtime = ?, pause_reg = ?
-        WHERE lc_package = ?
-      };
+  $self->assert_version_ok($ctx);
 
-      $Logger->log([
-        "updating packages: %s", {
-          package  => $package,
-          version  => $pp->{version},
-          dist     => $dist,
-          infile   => $pp->{infile},
-          filetime => $pp->{filemtime},
-          disttime => $self->dist->{TIME},
-        },
-      ]);
+  $Logger->log([
+    "updating packages: %s", {
+      package  => $package,
+      version  => $pp->{version},
+      dist     => $dist,
+      infile   => $pp->{infile},
+      filetime => $pp->{filemtime},
+      disttime => $self->dist->{TIME},
+    },
+  ]);
 
-      my $rows_affected = eval { $dbh->do
-                                     ($query,
-                                      undef,
-                                      $package,
-                                      lc $package,
-                                      $pp->{version},
-                                      $dist,
-                                      $pp->{infile},
-                                      $pp->{filemtime},
-                                      $self->dist->{TIME},
-                                      lc $package,
-                                     );
-                             };
-
-      if ($rows_affected) { # expecting only "1" can happen
-          $self->index_status
-              ($package,
-               $pp->{version},
-               $pp->{infile},
-               PAUSE::mldistwatch::Constants::OK,
-               "indexed",
-              );
-      } else {
-          my $dbherrstr = $dbh->errstr;
-          $self->index_status
-              ($package,
-               "undef",
-               $pp->{infile},
-               PAUSE::mldistwatch::Constants::EDBERR,
-               qq{The PAUSE indexer could not store the indexing
-result in the DB due the following error: C< $dbherrstr >.
-Please report the case to the PAUSE admins at modules\@perl.org.},
-              );
-      }
+  my $rows_affected = eval {
+      $dbh->do(
+          q{
+            UPDATE  packages
+            SET     package = ?, version = ?, dist = ?, file = ?,
```

Comment:
I think you lost an `lc_package = ?` ... `$package, lc $package, $pp->{version}` in a rebase

Reply:
I believe that `lc_package` can't change here.  Passing it in the update seems pointless.  We're finding the row for `lc_package = 'dist::zilla'` and we might update its `package` field to `DIST::ZiLLA` or `Dist::Zilla` or `Dist::zillA`, but no matter what `lc_package` has to remain constant.

Agreed?

Reply:
Yup